src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicConnection.cs

```diff
@@ -252,7 +252,7 @@ private static uint HandleEventShutdownComplete(State state, ref ConnectionEvent
             state.ShutdownTcs.SetResult(MsQuicStatusCodes.Success);
 
             // Stop accepting new streams.
-            state.AcceptQueue.Writer.Complete();
+            state.AcceptQueue.Writer.TryComplete();
```

Comment:
It seems like there are a bunch of places now where we either call Writer.Complete or Writer.TryComplete. Are all of these necessary? I would assume we only have to do this in a limited number of places, e.g. HandleEventShutdownInitiatedByTransport and HandleEventShutdownInitiatedByPeer, and maybe one other for when we initiate shutdown ourselves.

I always get nervous when we do stuff like call TryComplete instead of Complete, because it seems like we don't have a clean handling of exactly when the Complete operation needs to happen. I'd prefer to see the TryComplete calls changed to either Complete or to assert that it's already completed.

Reply:
In this case I was getting exception when when the connection was already Disposed and that triggered flush of events and ShutdownComplete.  I could wrap it with `if (_disposed == 0)` but changing it to `Try` look better to me. Since this only means we will not put more items to the queue.

Reply:
I look at the cases @geoffkizer and it is not that we would not know if the Complete operation should happen. We just don't know if happened before. Like in this case the shutdown may be initiated by peer or somebody may Dispose the connection but than we get this final event and it will throw if we try to complete it again. Unless I missed something I did not find API on the Writer to know if it was completed or not. 
One option I can think of you we making this nullable and set it to null after completion. Or we can add yet another state. Let me know what you want me to do.

Reply:
I think it's fine for this PR, but we should revisit the logic here moving forward.